yt/frontends/gdf/tests/test_outputs.py

```diff
@@ -1,30 +1,27 @@
-from yt.frontends.gdf.api import GDFDataset
-from yt.testing import assert_equal, requires_file, units_override_check
-from yt.utilities.answer_testing.framework import (
-    data_dir_load,
-    requires_ds,
-    small_patch_amr,
-)
+import pytest
 
-_fields = ("density", "velocity_x")
+from yt.frontends.gdf.api import GDFDataset
+from yt.testing import requires_file, units_override_check
+from yt.utilities.answer_testing.answer_tests import small_patch_amr
 
+# Test data
 sedov = "sedov/sedov_tst_0004.h5"
 
 
-@requires_ds(sedov)
-def test_sedov_tunnel():
-    ds = data_dir_load(sedov)
-    assert_equal(str(ds), "sedov_tst_0004")
-    for test in small_patch_amr(ds, _fields):
-        test_sedov_tunnel.__name__ = test.description
-        yield test
-
+@pytest.mark.answer_test
+class TestGDF:
+    answer_file = None
+    saved_hashes = None
 
-@requires_file(sedov)
-def test_GDFDataset():
-    assert isinstance(data_dir_load(sedov), GDFDataset)
+    @pytest.mark.usefixtures("hashing")
```

Comment:
Some of the marks we use show up over and over again. I think there's a way to reduce the boilerplate with config or something, isn't there?

Reply:
I didn't know about `pytest.mark.usefixtures`, but as I'm reading the docs I don't get why it's used here, as opposed to writing

```python
def test_sedov_tunnel(self, hashing, ...)
```

Reply:
IIRC, I did that because the hashing fixture doesn't actually return anything.  Additionally, it needs to do some work both before and after the test is run, so `yield` is employed. If there's a better way to do this, I'm open to it!

Reply:
I think there is, let me get back at you.

Reply:
@neutrinoceros Find anything? If not, no worries!

Reply:
Sorry for not coming back. I researched this a little and it's still quite confusing to me because the docs only mention this marker as useful to add fixtures to test classes.
(https://docs.pytest.org/en/stable/fixture.html#usefixtures)

This SO thread contains an answer that seems inline with your reasoning
https://stackoverflow.com/questions/50593556/pytest-how-to-use-a-mark-to-inject-a-fixture

TBH the fixture system in pytest is still half black magic to me, so I'm not too worried I don't really get this. Moving on !

Reply:
@neutrinoceros is this resolved?

Reply:
let's say it is. If I ever get a better grasp of this and decide this is bad practice, I'll propose a patch then.

Reply:
That sounds like a good plan!